src/utils/helper.ts

```diff
@@ -80,6 +97,18 @@ export function formatDecimalUnits(value: BigDecimal, decimals: BigInt): BigDeci
 	return value.div(scaler.toBigDecimal());
 }
 
+
+export function calcWeightedAverage(values: BigInt[], weights: BigInt[]): BigDecimal {
+
+	let numerator = BigInt.fromI32(0);
+	let denominator = BigInt.fromI32(0);
+	for (let i = 0; i < values.length; i++) {
+		denominator = denominator.plus(values[i].times(weights[i]))
+		numerator = numerator.plus(values[i]);
+	}
+	return numerator.toBigDecimal().div(denominator.toBigDecimal())
+}
```

Comment:
this maths looks right to me, but thoughts on this result being an int representation for consistency with other datapoints?

Reply:
it does eventually get turned into an int I guess I was just thinking of a more general calcWeightedAverage func